transhumandesign/kag-base
# Loading Raft on Catapult shows the correct icon, Raft blocks players from using actions

## Status

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

This is an addition/enhancement to [Pull Request #1215](https://github.com/transhumandesign/kag-base/pull/1215). You need to apply PR #1215 and then this to see the changes.

There are two changes. Please let me know what you think.

1)
Loading Raft on Catapult now shows the correct icon.
To do this, I edited VehicleIcons.png to add a small Raft sprite, replacing frame 7. (I did so because I think frame 7 was unused or was a placeholder anyway?)

2)
Players carrying a Raft will now be unable to use actions like slash, shield, bow, grapple(...)
Dinghy already has this implemented and I felt that Raft (which is supposed to be heavier weight) should implement this logic as well.
To do this, I added "BoatCommon.as" to "Raft.cfg".

## Steps to Test or Reproduce

- Try to load a Raft on a Catapult, notice it will use the correct sprite icon.
- Try to use actions while carrying a Raft, notice it will not work.
